src/jdk.compiler/share/classes/com/sun/tools/javac/comp/TransLiterals.java

```diff
@@ -0,0 +1,344 @@
+                if (iterator.hasNext()) {
+                    JCExpression expression = iterator.next();
+                    Type expressionType = expression.type;
+                    expr = makeBinary(Tag.PLUS, expr, expression.setType(expressionType));
+                }
+            }
+            return expr;
+        }
+
+        JCExpression bsmCall(Name name, Name bootstrapName, Type type,
+                             List<JCExpression> args,
+                             List<Type> argTypes,
+                             List<LoadableConstant> staticArgValues,
+                             List<Type> staticArgsTypes) {
+            Symbol bsm = rs.resolveQualifiedMethod(tree.pos(), env,
+                    syms.templateRuntimeType, bootstrapName, staticArgsTypes, List.nil());
+            MethodType indyType = new MethodType(argTypes, type, List.nil(), syms.methodClass);
+            DynamicMethodSymbol dynSym = new DynamicMethodSymbol(
+                    name,
+                    syms.noSymbol,
+                    ((MethodSymbol)bsm).asHandle(),
+                    indyType,
+                    staticArgValues.toArray(new LoadableConstant[0])
+            );
+            JCFieldAccess qualifier = make.Select(make.Type(syms.processorType), dynSym.name);
+            qualifier.sym = dynSym;
+            qualifier.type = type;
+            JCMethodInvocation apply = make.Apply(List.nil(), qualifier, args);
+            apply.type = type;
+            return apply;
+        }
+
+        JCExpression processCall(JCExpression stringTemplate) {
+            MethodSymbol appyMeth = lookupMethod(tree.pos(), names.process,
+                    syms.processorType, List.of(syms.stringTemplateType));
+            JCExpression applySelect = make.Select(processor, appyMeth);
+            JCExpression process = make.Apply(null, applySelect, List.of(stringTemplate))
+                    .setType(syms.objectType);
+            JCTypeCast cast = make.TypeCast(tree.type, process);
+            return cast;
+        }
+
+        JCExpression newStringTemplate() {
+            List<LoadableConstant> staticArgValues = List.nil();
+            List<Type> staticArgsTypes =
+                    List.of(syms.methodHandleLookupType, syms.stringType,
+                            syms.methodTypeType);
+            if (useValuesList) {
+                JCNewArray fragmentArray = make.NewArray(make.Type(syms.stringType),
+                        List.nil(), makeStringList(fragments));
+                fragmentArray.type = new ArrayType(syms.stringType, syms.arrayClass);
+                JCNewArray valuesArray = make.NewArray(make.Type(syms.objectType),
+                        List.nil(), expressions);
+                valuesArray.type = new ArrayType(syms.objectType, syms.arrayClass);
+                return bsmCall(names.process, names.newLargeStringTemplate, syms.stringTemplateType,
```

Comment:
nit: the indy name is irrelevant here - that said, `process` is a tad confusing, since it's also the name of a StringTemplate method.

Reply:
Tracks well in javah